examples/bots/discord-bot/src/main.rs

```diff
@@ -0,0 +1,170 @@
+struct App {
+    #[clap(
+        long = "token",
+        help = "Discord bot API token.",
+        env = "DISCORD_BOT_TOKEN"
+    )]
+    bot_token: String,
+    #[clap(
+        long = "log-level",
+        default_value = "info",
+        help = "Maximum log level.",
+        env = "DISCORD_BOT_LOG_LEVEL"
+    )]
+    log_level: tracing_subscriber::filter::LevelFilter,
+    #[clap(
+        long = "verifier-url",
+        default_value = "http://127.0.0.1:8080/",
+        help = "URL of the SoMe verifier.",
+        env = "SOME_VERIFIER_URL"
+    )]
+    verifier_url: Url,
+}
+
+struct BotConfig {
+    verifier_url: Url,
+    client: reqwest::Client,
+}
+type Context<'a> = poise::Context<'a, BotConfig, anyhow::Error>;
+
+// Note: The doc comments of the commands below define what the user sees as a help message
+/// Displays the list of commands
+#[poise::command(slash_command, prefix_command)]
+async fn help(
+    ctx: Context<'_>,
+    #[description = "Specific command to show help about"]
+    #[autocomplete = "poise::builtins::autocomplete_command"]
+    command: Option<String>,
+) -> anyhow::Result<()> {
+    poise::builtins::help(
+        ctx,
+        command.as_deref(),
+        poise::builtins::HelpConfiguration::default(),
+    )
+    .await?;
+    Ok(())
+}
+
+/// Verify with Concordium
+#[poise::command(slash_command, prefix_command)]
+async fn verify(ctx: Context<'_>) -> anyhow::Result<()> {
+    ctx.send(|reply| {
+        reply
+            .content("Please verify with your wallet.")
+            .components(link_button(&ctx.data().verifier_url, "Verify"))
+            // Only the recipient can see the message
+            .ephemeral(true)
+    })
+    .await?;
+    Ok(())
+}
+
+/// Checks the verification status of a user
+#[poise::command(slash_command, prefix_command)]
+async fn check(
+    ctx: Context<'_>,
+    #[description = "Selected user"] user: serenity::User,
+) -> anyhow::Result<()> {
+    match get_verifications(ctx.data(), user.id).await {
+        Ok(verifications) => {
+            let mention = user.mention();
+            let message = if verifications.is_empty() {
+                format!("{mention} is not verified with Concordium.")
+            } else {
+                let mut message = format!("{mention} is verified with Concordium.");
+                for verification in verifications
+                    .into_iter()
+                    .filter(|v| v.platform != Platform::Discord && !v.revoked)
+                {
+                    message.push_str(&format!(
+                        "\n- {}: {}",
+                        verification.platform, verification.username
+                    ));
+                }
+                message
+            };
+            // .ephemeral(true) means only the recipient can see the message
+            ctx.send(|reply| reply.content(message).ephemeral(true))
+                .await?;
+        }
+        Err(err) => tracing::error!("{err}"),
+    }
+
+    Ok(())
```

Comment:
What is the meaning of this OK response? It seems we ignore the error if we cannot get verifications, which is an odd choice.